tests/unit/test_download.py

```diff
@@ -51,7 +51,27 @@ def _fake_session_get(*args, **kwargs):
 
 
 def test_user_agent():
-    PipSession().headers["User-Agent"].startswith("pip/%s" % pip.__version__)
+    user_agent = PipSession().headers["User-Agent"]
+
+    assert user_agent.startswith("pip/%s" % pip.__version__)
+
+
+@pytest.mark.parametrize('name, expected_like_ci', [
+    ('BUILD', False),
+    ('BUILD_BUILDID', True),
+    ('BUILD_ID', True),
+    ('CI', True),
+])
+def test_user_agent__ci(monkeypatch, name, expected_like_ci):
+    # Clear existing names since we can be running under an actual CI.
+    for ci_name in ('BUILD_BUILDID', 'BUILD_ID', 'CI'):
```

Comment:
this should probably be the array you defined above, to not have it as duplicate and auto-extend in future